torch/nn/modules/rnn.py

```diff
@@ -252,11 +253,12 @@ def flatten_parameters(self) -> None:
         # a sufficient check, because overlapping parameter buffers that don't completely
         # alias would break the assumptions of the uniqueness check in
         # Module.named_parameters().
-        unique_data_ptrs = {
-            p.data_ptr()  # type: ignore[union-attr]
-            for p in self._flat_weights
+        unique_storage_refs = {
+            StorageWeakRef(p.untyped_storage())
```

Comment:
Why move from data_ptr() to untyped_storage() here if the new method is added to expanded_weights either way?